### PR TITLE
Add a few unit tests

### PR DESCRIPTION
**Pull request by qbarrand**
This PR adds a few unit tests. In contrary to the end-to-end tests located under the `test` directory, they are self-contained and do not need a cluster to run.
The `make test` target now runs unit tests and not the end-to-end tests, thanks to build tags. To run the full suite, the `make test-e2e` target can be used.

### CODE REVIEW

File: pkg/assets/assets_test.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import (
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestValidStateName(t *testing.T) {
+	cases := []struct {
+		input string
+		valid bool
+	}{
```
Comment (pacevedom) [inline pkg/assets/assets_test.go:12]: I am missing a test with more than 4 digits.

File: pkg/conditions/conditions_test.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	assert.Equal(t, "Reconciled", progressing.Reason)
+	assert.Equal(t, "SpecialResources up to date", progressing.Message)
+	assert.False(t, progressing.LastTransitionTime.IsZero())
+
+	degraded := v1helpers.FindStatusCondition(conds, configv1.OperatorDegraded)
+	assert.NotNil(t, degraded)
+	assert.Equal(t, configv1.ConditionFalse, degraded.Status)
+	assert.Equal(t, "AsExpected", degraded.Reason)
+	assert.Equal(t, "Special Resource Operator reconciling special resources", degraded.Message)
+	assert.False(t, degraded.LastTransitionTime.IsZero())
```
Comment (pacevedom) [inline pkg/conditions/conditions_test.go]: Maybe adding these in a []struct{...} and iterating?

File: pkg/conditions/conditions_test.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+	assert.Equal(t, "Reconciling", progressing.Reason)
+	assert.Equal(t, msgProgressing, progressing.Message)
+	assert.False(t, progressing.LastTransitionTime.IsZero())
+
+	degraded := v1helpers.FindStatusCondition(conds, configv1.OperatorDegraded)
+	assert.NotNil(t, degraded)
+	assert.Equal(t, configv1.ConditionFalse, degraded.Status)
+	assert.Equal(t, "Reconciled", degraded.Reason)
+	assert.Equal(t, msgDegraded, degraded.Message)
+	assert.False(t, degraded.LastTransitionTime.IsZero())
```
Comment (pacevedom) [inline pkg/conditions/conditions_test.go]: Same comment.

File: pkg/kernel/kernel_test.go [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"testing"
+)
+
+const kernelFullVersion = "4.18.0-305.19.1.el8_4.x86_64"
+
+func TestSetAffineAttributes(t *testing.T) {
+	const (
+		objName                   = "test-obj"
+		objNameHash               = "bfb16b50984f16f0" // fnv64a(operatingSystemMajorMinor + kernelFullVersion)
```
Comment (pacevedom) [inline pkg/kernel/kernel_test.go:23]: Maybe generate this based on constants and then calling fnv64a?

File: pkg/kernel/kernel_test.go [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+
+	newObj := func(kind string) *unstructured.Unstructured {
+		obj := unstructured.Unstructured{}
+		obj.SetName(objName)
+		obj.SetKind(kind)
+
+		return &obj
+	}
+
+	t.Run("BuildRun", func(t *testing.T) {
```
Comment (pacevedom) [inline pkg/kernel/kernel_test.go:28]: What is a BuildRun?

File: pkg/kernel/kernel_test.go [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+				require.NoError(t, err)
+				assert.True(t, ok)
+				assert.Equal(t, expectedSelector, m)
+			}
+		})
+	}
+}
+
+func TestSetVersionNodeAffinity(t *testing.T) {
+	newObj := func(kind string) *unstructured.Unstructured {
```
Comment (pacevedom) [inline pkg/kernel/kernel_test.go]: This is defined twice, maybe move it to unexported function?

**Review by pacevedom — COMMENTED** [review]
Few nits

File: pkg/kernel/kernel_test.go (hunk 5 above)
Comment (qbarrand) [reply pkg/kernel/kernel_test.go:28]: It looks to me like [Shipwright's equivalent to OpenShift's BuildConfig](https://github.com/shipwright-io/build/blob/main/docs/buildrun.md).

**Comment by ybettan** [comment]
Very nice, maybe wort adding this testing flow to the CI as well.

**Comment by qbarrand** [comment]
@ybettan the `test` target used to be skipped because it had no dependencies; I added `patch`, so now the unit tests are ran during `make local-image-build`.

File: pkg/kernel/kernel_test.go (hunk 4 above)
Comment (qbarrand) [reply pkg/kernel/kernel_test.go:23]: This would replicate the logic in the code that is tested, so I'm not sure I would go that route.

**Review by pacevedom — APPROVED** [review]

**Pull request merged**